File: mipush_hook/src/main/java/com/nihility/DefaultHookedMethodHandler.java
package com.nihility;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;

import com.xiaomi.network.Fallback;
import com.xiaomi.push.service.MIPushNotificationHelper;
import com.xiaomi.push.service.XMPushService;
import com.xiaomi.xmpush.thrift.PushMetaInfo;
import com.xiaomi.xmpush.thrift.XmPushActionContainer;

import org.aspectj.lang.JoinPoint;
import org.aspectj.lang.ProceedingJoinPoint;

class DefaultHookedMethodHandler implements HookedMethodHandler {
    @Override
    public boolean shouldSendBroadcast(ProceedingJoinPoint joinPoint, XMPushService pushService, String packageName, XmPushActionContainer container, PushMetaInfo metaInfo) throws Throwable {
        return (boolean) joinPoint.proceed();
    }

    @Override
    public void postProcessMIPushMessage(ProceedingJoinPoint joinPoint, XMPushService pushService, String pkgName, byte[] payload, Intent newMessageIntent) throws Throwable {
        joinPoint.proceed();
    }

    @Override
    public void notifyPacketArrival(JoinPoint joinPoint, XMPushService pushService, String chid, Object data) {

    }

    @Override
    public Object debugLog(ProceedingJoinPoint joinPoint) throws Throwable {
        return joinPoint.proceed();
    }

    @Override
    public void logFallback(JoinPoint joinPoint, Fallback fallback, boolean usePort) {

    }

    @Override
    public void processIntent(JoinPoint joinPoint, Intent intent) {

    }

    @Override
    public void onCreate(JoinPoint joinPoint, XMPushService pushService) throws Throwable {

    }

    @Override
    public void onStartCommand(JoinPoint joinPoint) {

    }

    @Override
    public void onStart(JoinPoint joinPoint, Intent intent, int startId) {

    }

    @Override
    public void onBind(JoinPoint joinPoint, Intent intent) {

    }

    @Override
    public void onDestroy(JoinPoint joinPoint) {

    }

    @Override
    public void setConnectionStatus(JoinPoint joinPoint, int newStatus, int reason, Exception e) {

    }

    @Override
    public void sendMessage(JoinPoint joinPoint, Intent intent) {

    }

    @Override
    public void logCheckServices(JoinPoint joinPoint, PackageInfo pkgInfo) {

    }

    @Override
    public Intent buildIntent(ProceedingJoinPoint joinPoint) throws Throwable {
        return (Intent) joinPoint.proceed();
    }

    @Override
    public XmPushActionContainer buildContainerHook(ProceedingJoinPoint joinPoint) throws Throwable {
        return (XmPushActionContainer) joinPoint.proceed();
    }

    @Override
    public boolean isIntentAvailable(ProceedingJoinPoint joinPoint) throws Throwable {
        return (boolean) joinPoint.proceed();
    }

    @Override
    public void processMIPushMessage(JoinPoint joinPoint, XMPushService pushService, byte[] decryptedContent, long packetBytesLen) {

    }

    @Override
    public boolean isDuplicateMessage(ProceedingJoinPoint joinPoint, XMPushService pushService, String packageName, String messageId) throws Throwable {
        return (boolean) joinPoint.proceed();
    }

    @Override
    public MIPushNotificationHelper.NotifyPushMessageInfo notifyPushMessage(ProceedingJoinPoint joinPoint, Context context, XmPushActionContainer container, byte[] decryptedContent) throws Throwable {
        return (MIPushNotificationHelper.NotifyPushMessageInfo) joinPoint.proceed();
    }
}

File: mipush_hook/src/main/java/com/nihility/DefaultHookedMethodHandler.kt
package com.nihility

import android.content.Context
import android.content.Intent
import android.content.pm.PackageInfo
import com.xiaomi.network.Fallback
import com.xiaomi.push.service.MIPushNotificationHelper
import com.xiaomi.push.service.XMPushService
import com.xiaomi.xmpush.thrift.PushMetaInfo
import com.xiaomi.xmpush.thrift.XmPushActionContainer
import org.aspectj.lang.JoinPoint
import org.aspectj.lang.ProceedingJoinPoint

class DefaultHookedMethodHandler : HookedMethodHandler {
    override fun shouldSendBroadcast(
        joinPoint: ProceedingJoinPoint,
        pushService: XMPushService,
        packageName: String,
        container: XmPushActionContainer,
        metaInfo: PushMetaInfo
    ): Boolean = joinPoint.proceed() as Boolean

    override fun postProcessMIPushMessage(
        joinPoint: ProceedingJoinPoint,
        pushService: XMPushService,
        pkgName: String,
        payload: ByteArray,
        newMessageIntent: Intent
    ) {
        joinPoint.proceed()
    }

    override fun notifyPacketArrival(joinPoint: JoinPoint, pushService: XMPushService, chid: String, data: Any) {}
    override fun debugLog(joinPoint: ProceedingJoinPoint): Any? = joinPoint.proceed()
    override fun logFallback(joinPoint: JoinPoint, fallback: Fallback, usePort: Boolean) {}
    override fun processIntent(joinPoint: JoinPoint, intent: Intent) {}
    override fun onCreate(joinPoint: JoinPoint, pushService: XMPushService) {}
    override fun onStartCommand(joinPoint: JoinPoint) {}
    override fun onStart(joinPoint: JoinPoint, intent: Intent, startId: Int) {}
    override fun onBind(joinPoint: JoinPoint, intent: Intent) {}
    override fun onDestroy(joinPoint: JoinPoint) {}
    override fun setConnectionStatus(joinPoint: JoinPoint, newStatus: Int, reason: Int, e: Exception) {}
    override fun sendMessage(joinPoint: JoinPoint, intent: Intent) {}
    override fun logCheckServices(joinPoint: JoinPoint, pkgInfo: PackageInfo) {}
    override fun buildIntent(joinPoint: ProceedingJoinPoint): Intent = joinPoint.proceed() as Intent
    override fun buildContainerHook(joinPoint: ProceedingJoinPoint): XmPushActionContainer =
        joinPoint.proceed() as XmPushActionContainer

    override fun isIntentAvailable(joinPoint: ProceedingJoinPoint): Boolean = joinPoint.proceed() as Boolean

    override fun processMIPushMessage(
        joinPoint: JoinPoint,
        pushService: XMPushService,
        decryptedContent: ByteArray,
        packetBytesLen: Long
    ) {
    }

    override fun isDuplicateMessage(
        joinPoint: ProceedingJoinPoint,
        pushService: XMPushService,
        packageName: String,
        messageId: String
    ): Boolean = joinPoint.proceed() as Boolean

    override fun notifyPushMessage(
        joinPoint: ProceedingJoinPoint,
        context: Context,
        container: XmPushActionContainer,
        decryptedContent: ByteArray
    ): MIPushNotificationHelper.NotifyPushMessageInfo =
        joinPoint.proceed() as MIPushNotificationHelper.NotifyPushMessageInfo
}
